### DISCUSSION TITLE
Upgrade 13.3.4 -> 13.4.1: TypeError: Cannot read properties of null (reading 'useContext')

### DISCUSSION THREAD

**Discussion by MikeEnRegalia** [vercel/next.js#50525] [Help]
### Summary

This cost me several hours ... first I thought I did something wrong, but this keeps happening when I upgrade my app and try to build it. Any thoughts? The usual suspect is conflicting react versions, but I'm using 18.20 and there is no conflicting configuration in my project. Have also tried cleaning node_modules as well as deleting package-lock.json - no luck.

### Additional information

This is happening in the old pages/ folder, not in the new apps/ folder.

**Comment by icyJoseph** [comment]
This can also happen when accessing Providers within Server Components. 

Did you change anything at all?

**Reply by MikeEnRegalia** [reply]
Good point. How do I know whether I'm in a server component? In my case, it seems to be a useSWR() call that triggers the problem. The call is in a component which is in a plain Page. Shouldn't that work? I'm not aware of any need to check whether I'm rendering server or client side before I can useSWR() ... I thought that the hook automatically only works client side.

**Reply by icyJoseph** [reply]
Everything is a server component, unless... you mark it as "use client", or if its part of a "use client" tree. The second option is what I've found most people trip with, but it generally breaks the app so you can figure out what is going on.

If you say plain Page, I understand a page with "use client" directive, which makes it a Server Component.

**Reply by MikeEnRegalia** [reply]
Even in old Pages? I'm not using the apps folder yet ...

**Reply by icyJoseph** [reply]
Oh, that shouldn't impact you then. Pages Directory should be alright.

If you haven't done any other code changes, just upgrading Next.js, it might be a bug.

**Reply by icyJoseph** [reply]
Perhaps: https://github.com/vercel/next.js/issues/49382 ?

**Reply by MikeEnRegalia** [reply]
Maybe - but that's also for the apps directory ...
